File: android/src/main/java/expo/modules/mocklivephoto/ExpoMockLivePhotoModule.kt
package expo.modules.mocklivephoto

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoMockLivePhotoModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoMockLivePhoto")

    View(ExpoMockLivePhotoView::class) {
      Events("onVideoReady", "onPlaybackStart", "onPlaybackPause", "onPlaybackEnd", "onError")

      Prop("videoUri") { view: ExpoMockLivePhotoView, value: String -> view.setVideoUri(value) }
      Prop("muted") { view: ExpoMockLivePhotoView, value: Boolean -> view.setMuted(value) }
      Prop("resizeMode") { view: ExpoMockLivePhotoView, value: String -> view.setResizeMode(value) }

      AsyncFunction("play") { view: ExpoMockLivePhotoView -> view.play() }
      AsyncFunction("pause") { view: ExpoMockLivePhotoView -> view.pause() }
      AsyncFunction("reset") { view: ExpoMockLivePhotoView -> view.reset() }

      OnViewDestroys { view -> view.reset() }
    }
  }
}

File: ios/ExpoMockLivePhotoModule.swift
import ExpoModulesCore

public class ExpoMockLivePhotoModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoMockLivePhoto")

    View(ExpoMockLivePhotoView.self) {
      Events("onVideoReady", "onPlaybackStart", "onPlaybackEnd", "onError")

      Prop("videoUri") { (view, value: String) in
        view.setVideoUri(value)
      }
      Prop("muted") { (view, value: Bool) in
        view.setMuted(value)
      }
      Prop("resizeMode") { (view, value: String) in
        view.setResizeMode(value)
      }

      AsyncFunction("play") { (view: ExpoMockLivePhotoView) in
        view.play()
      }
      AsyncFunction("pause") { (view: ExpoMockLivePhotoView) in
        view.pause()
      }
      AsyncFunction("reset") { (view: ExpoMockLivePhotoView) in
        view.reset()
      }
    }
  }
}
